### DISCUSSION TITLE
How to tell deck.gl that "My layer's extensions has changed"?

### DISCUSSION THREAD

**Discussion by lianzhao** [visgl/deck.gl#6942] [Q&A] [answered]
I want to tell deck.gl that "My layer's extensions has changed" when `myVariable` changed. My code below not working.
```js
const layers = [{
  // ...
  extensions: [
    new MyExtension(myVariable), // will overwrite getShaders method in my extension
  ],
  updateTriggers: {
    extensions: myVariable,
    getShaders: myVariable,
  },
}];
```

**Comment by Pessimistress** [comment, marked as answer]
It should just work without any update triggers (update trigger is for get* accessors).

The `LayerExtension` class has a `equals(otherExtension): boolean` method. The default implementation deep compares the opts passed to both constructors.

**Reply by lianzhao** [reply]
> The `LayerExtension` class has a `equals(otherExtension): boolean` method. The default implementation deep compares the opts passed to both constructors.

@Pessimistress Thank you! I changed my constructor from
```javascript
constructor(opts) {
  const { myVariable, ...rest} = opts;
  super(rest);
}
```
to 
```javascript
constructor(opts) {
  const { myVariable, ...rest} = opts;
  super(opts); // 'rest' did not change...
}
```
and it worked as you said. 😀
